src/isomorphic/devtools/ReactComponentTreeDevtool.js

```diff
@@ -106,6 +106,11 @@ var ReactComponentTreeDevtool = {
   },
 
   purgeUnmountedComponents() {
+    if (ReactComponentTreeDevtool._preventPurging) {
```

Comment:
I’d keep it this way for now. I’m not sure about this reliance on flush events anyway (https://github.com/facebook/react/pull/6046#discussion_r61482538), so I’d rather avoid relying on the order in which devtools are called for now.